adding general fusaka upgrade page (#16135)
ethereum/ethereum-org-website: public/content/roadmap/fusaka/index.md

---
title: Fulu-Osaka (Fusaka)
description: Learn about the Fusaka protocol upgrade
lang: en
---

# Fusaka {#fusaka}

> The Fusaka network upgrade follows [Pectra](/roadmap/pectra/) and brings more new features and improves the experience for every Ethereum user and developer. The name consists of execution layer upgrade Osaka and consensus layer version named after the Fulu star. Both parts of Ethereum receive an upgrade that push Ethereum scaling, security and user experience to the future. 

Reviewer: ```suggestion
The Fusaka network upgrade follows [Pectra](/roadmap/pectra/) and brings more new features and improves the experience for every Ethereum user and developer. The name consists of the execution layer upgrade Osaka and the consensus layer version named after the Fulu star. Both parts of Ethereum receive an upgrade that pushes Ethereum scaling, security and user experience to the future. 
```

This upgrade is planned for Q4 2025.

<InfoBanner>
The Fusaka upgrade is only a single step in Ethereum's long-term development goals. Learn more about [the protocol roadmap](/roadmap/) and [previous upgrades](/history/).
</InfoBanner>

## Improvements in Fusaka {#new-improvements}

### Data availability & L2 scaling {#da-scaling}

#### PeerDAS {#7594}

Specification: https://eips.ethereum.org/EIPS/eip-7594

Resources: https://youtu.be/bONWd1x2TjQ?t=328 (dapplion on PeerDAS)

This is the *headliner* of the Fusaka fork, the main feature added in this upgrade. L2s currently post their data to Ethereum in blobs, the ephemeral data type created specifically for L2s. Pre-Fusaka, every full node has to store every **byte of those blobs to ensure that the data exists. As blob throughput rises, having to download all of this data would be untenably resource-intensive.

> With [Data Availability Sampling](https://notes.ethereum.org/@fradamt/das-fork-choice) , instead of having to download all of the blob data, each node will sample just a portion. Blobs are uniformly randomly distributed across nodes in the network with each full node holding only 1/8th of the data, therefore enabling theoretical scale up to 8x. To ensure availability of the data,  Any portion of the data can be reconstructed from any existing 50% of the whole with methods that drive down the probability of wrong or missing data to a cryptographically negligible level (~one in 10²⁰ to one in 10²⁴).

Reviewer: ```suggestion
With [Data Availability Sampling](https://notes.ethereum.org/@fradamt/das-fork-choice), instead of having to download all of the blob data, each node will sample just a portion. Blobs are uniformly randomly distributed across nodes in the network, with each full node holding only 1/8th of the data, therefore enabling a theoretical scale up to 8x. To ensure availability of the data, it's represented by a polynomial with a cryptographic proof that enables reconstruction of any piece of the original data from only 50% of available blobs using erasure coding. 
```

This keeps hardware and bandwidth requirements for nodes tenable while enabling blob scaling resulting in more scale with smaller fees for L2s.

In-depth: https://eprint.iacr.org/2024/1362.pdf

#### Blob parameter only forks

Specification: https://eips.ethereum.org/EIPS/eip-7892

L2s scale Ethereum - as their networks grow, they need to post more data to Ethereum. This means that Ethereum will need to increase the number of blobs available to them as time goes on. Although PeerDAS enables scaling blob data, it needs to be done in gradually and safely. 

> Because Ethereum is code running on thousands of independent nodes that requirement agreement on the rules, you can’t just make changes like blob increases the way you deploy a website update. Any rule change must be a coordinated upgrade where every validator software upgrades at the same predetermined block.

Reviewer: ```suggestion
Because Ethereum is code running on thousands of independent nodes that require agreement on same rules, we cannot simply introduces changes like increasing blob count the way you deploy a website update. Any rule change must be a coordinated upgrade where every node, client and validator software upgrades before the same predetermined block.
```

> These coordinated upgrades generally include a lot of changes, require a lot of testing, and that takes time. In order to adapt faster to changing L2 blob needs, BPO-only forks introduce a mechanism to increase blobs without having to wait on that upgrade schedule.

Author: idk how to edit granularly but "parameter" misspelled

> BPO-only forks can be set by clients. Between major ethereum upgrades, clients can agree to increase the `target` and `max` blobs to e.g. 9 and 12 and then node operators will update to take part in that tiny fork. These BPO-only forks can be configured at any time.

Reviewer: ```suggestion
BPO-only forks can be set by clients, similarly to other configuration like gas limit. Between major Ethereum upgrades, clients can agree to increase the `target` and `max` blobs to e.g. 9 and 12 and then node operators will update to take part in that tiny fork. These BPO-only forks can be configured at any time.
```

#### Set default gas limit to XX million

Specification: https://eips.ethereum.org/EIPS/eip-7935

Prior to raising the gas limit from 30M to 36M in Feb 2025 (and subsequently to 45M), this value hadn’t changed since the Merge (Sep 2022). This EIP aims to make consistent scaling a priority.

EIP-7935 coordinates EL client teams to raise the default gas-limit above today’s 45M for Fusaka. It’s an Informational EIP, but it explicitly asks clients to test higher limits on devnets, converge on a safe value, and ship that number in their Fusaka releases.

Devnet planning targets ~60M stress (full blocks with synthetic load) and iterative bumps; research says worst-case block-size pathologies shouldn’t bind below ~150M. Rollout should be paired with the transaction gas-limit cap (EIP-7825) so no single transaction can dominate as limits rise.

### Preconfirmation support

#### Deterministic proposer lookahead

Specification: https://eips.ethereum.org/EIPS/eip-7917

> With EIP-7917, Beacon Chain will become aware of upcoming block proposers for next epoch. Having a deterministic view on which validators will be propose future blocks can enable [preconfirmations](https://ethresear.ch/t/based-preconfirmations/17353) - a commitment with upcoming proposer that guarantees the user transaction will be included in their block without waiting for the actual block. 

Reviewer: ```suggestion
With EIP-7917, Beacon Chain will become aware of upcoming block proposers for the next epoch. Having a deterministic view on which validators will be proposing future blocks can enable [preconfirmations](https://ethresear.ch/t/based-preconfirmations/17353) - a commitment with the upcoming proposer that guarantees the user transaction will be included in their block without waiting for the actual block. 
```
